coordinator/src/writer.rs

```diff
@@ -395,17 +395,23 @@ impl PointWriter {
                 span_recorder.span_ctx(),
             )
             .await;
-        if let Err(err @ CoordinatorError::FailoverNode { .. }) = result {
-            debug!(
-                "write data to remote {}({}) failed; write to hinted handoff!",
-                node_id, vnode_id
-            );
-
-            span_recorder.error(err.to_string());
 
-            return self
-                .write_to_handoff(vnode_id, node_id, tenant, precision, data)
-                .await;
+        if let Err(ref err) = result {
+            if matches!(*err, CoordinatorError::FailoverNode { .. })
+                || err.error_code().to_string() == *"020005"
```

Comment:
use TSKV::*  Error ？

Reply:
Errors are nested, and the error_code method can unnest them